File: Chapter 14/JS/section_14_6_instrumentation_logging/ports/happyPathPaymentPort.js
/**
 * A concrete, hand-coded stub implementation of PaymentPort.
 * Simulates a successful payment with zero network overhead.
 */
import { PaymentPort } from "./paymentPort.js";

export class HappyPathPaymentPort extends PaymentPort {
  /** @param {number} amount */
  process(amount) {
    return true;
  }
}

File: Chapter 14/Python/section_14_6_instrumentation_logging/ports/happy_path_payment_port.py
"""
A concrete, hand-coded stub implementation of PaymentPort.
Simulates a successful payment with zero network overhead.
"""

from section_14_6_instrumentation_logging.ports.payment_port import PaymentPort


class HappyPathPaymentPort(PaymentPort):
    """Hand-coded stub that always returns success."""

    def process(self, amount: float) -> bool:
        return True